Mathlib/Analysis/SpecialFunctions/Log/Base.lean

```diff
@@ -427,3 +441,29 @@ theorem logb_prod {α : Type*} (s : Finset α) (f : α → ℝ) (hf : ∀ x ∈
 #align real.logb_prod Real.logb_prod
 
 end Real
+
+section Induction
+
+/-- Induction principle for intervals of real numbers: if a proposition `P` is true
+on `[x₀, r * x₀)` and if `P` for `[x₀, r^n * x₀)` implies `P` for `[r^n * x₀, r^(n+1) * x₀)`,
+then `P` is true for all `x ≥ x₀`. -/
+lemma Real.induction_Ico_mul {P : ℝ → Prop} (x₀ r : ℝ) (hr : 1 < r) (hx₀ : 0 < x₀)
+    (base : ∀ x ∈ Set.Ico x₀ (r * x₀), P x)
+    (step : ∀ n : ℕ, n ≥ 1 → (∀ z ∈ Set.Ico x₀ (r ^ n * x₀), P z) →
+      (∀ z ∈ Set.Ico (r ^ n * x₀) (r ^ (n+1) * x₀), P z)) :
+    ∀ x ≥ x₀, P x := by
```

Comment:
The additive version can be generalized to the following:
```
import Mathlib.Algebra.Order.Archimedean

lemma induction_Ico_add_nsmul {α} [LinearOrderedAddCommGroup α] [Archimedean α]
    {P : α → Prop} (x₀ r : α) (hr : 0 < r)
    (base : ∀ x ∈ Set.Ico x₀ (x₀ + r), P x)
    (step : ∀ n ≥ 1, (∀ z ∈ Set.Ico x₀ (x₀ + n • r), P z) →
      (∀ z ∈ Set.Ico (x₀ + n • r) (x₀ + (n+1) • r), P z)) :
    ∀ x ≥ x₀, P x := by
  sorry
```
though trying to transfer that result may lead to a longer proof.

Reply:
Yeah, and I think this doesn't need anything about logs for its proof. So while this might be valuable, I don't think it needs to be part of this PR.

Reply:
Right -- I'd rather avoid adding it to this PR, this is already a rabbit hole I went down in order to generalize lemmas in another PR.